Podman regression in Ubuntu 22.04 image released Feb 25th?

### Description

I think there might be a regression in [this release](https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240225.1) of the ubuntu-22.04 image which breaks podman.

The image updated the kernel from 6.2.y to 6.5.y, but podman/crun don't seem to be updated.

Our build fails with this error [link to run](https://github.com/gardenlinux/gardenlinux/actions/runs/8061893233/job/22020522535):

```
STEP 1/1: FROM ghcr.io/gardenlinux/builder:3ab2bb52bc46bb200c761369c087e9413d1ce0ac
Trying to pull ghcr.io/gardenlinux/builder:3ab2bb52bc46bb200c761369c087e9413d1ce0ac...
Getting image source signatures
Copying blob sha256:041b542221cfde2f9fa4ac13f8b5804e25b23ab48ba47db2822c382a134256e1
Copying blob sha256:041b542221cfde2f9fa4ac13f8b5804e25b23ab48ba47db2822c382a134256e1
Copying config sha256:1eba10d0345cc6df78b7c3a6ced45da9db675d05eb20d5d286996e4f7ffb24d5
Writing manifest to image destination
Storing signatures
COMMIT localhost/builder
--> 1eba10d0345
Successfully tagged localhost/builder:latest
Successfully tagged ghcr.io/gardenlinux/builder:3ab2bb52bc46bb200c761369c087e9413d1ce0ac
1eba10d0345cc6df78b7c3a6ced45da9db675d05eb20d5d286996e4f7ffb24d5
Error: OCI runtime error: chmod `run/shm`: Operation not supported
Error: Process completed with exit code 126.
```

This is with this image version:

```
Current runner version: '2.313.0'
Operating System
  Ubuntu
  22.04.4
  LTS
Runner Image
  Image: ubuntu-22.04
  Version: 20240225.1.0
  Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240225.1/images/ubuntu/Ubuntu2204-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240225.1
```

Trying to reproduce, it seems like I am only able to get this image version `20240218.1.0` where the issue does not appear.

Tried to reproduce [in this repo](https://github.com/fwilhe/turbo-lamp/blob/main/.github/workflows/podman.yml), but I'm not able to get this with image version `20240225.1.0`.

Is this a known issue and version `20240225.1.0` is not in use anymore?

[This blog post](https://noobient.com/2023/11/15/fixing-ubuntu-containers-failing-to-start-with-systemd/) seems to suggest that the crun version is too old.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Fails with: `20240225.1.0`
Works with: `20240218.1.0`

Failing build: https://github.com/gardenlinux/gardenlinux/actions/runs/8061893233/job/22020522535

### Is it regression?

yes

### Expected behavior

Podman build should work

### Actual behavior

Podman build fails with message:
```
Error: OCI runtime error: chmod `run/shm`: Operation not supported
```

### Repro steps

Create a workflow running on ubuntu 22.04 and execute those steps:

```
    - name: install required packages
      shell: bash
      run: |
        sudo rm -rf /{usr/{local/{lib/{android,heroku},.ghcup,share/{dotnet,powershell,miniconda,swift}},share/{dotnet,miniconda,swift}},opt/{hostedtoolcache,microsoft},imagegeneration}
        sudo docker system prune -a -f
        sudo apt-get update
        sudo env DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends qemu-user-static

    - run: |
        git clone https://github.com/gardenlinux/gardenlinux
        cd gardenlinux
        ./build base-amd64 base-arm64
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -9,7 +9,7 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
       - name: Build the image

